envoy/service/auth/v2/external_auth.proto

```diff
@@ -26,4 +26,22 @@ message CheckRequest {
 message CheckResponse {
   // Status `OK` allows the request. Any other status indicates the request should be denied.
   google.rpc.Status status = 1;
+
+  // An optional message which contains the attributes of an HTTP response object. This message is
+  // particularly useful when the authorization service needs to send specific responses to the
+  // downstream client or to modify the request headers before dispatching them to the upstream.
+  // E.g., when the authorization status is not OK, this message can be used by the authorization
+  // service to the tell the filter to respond with specific HTTP headers, body, and status code.
+  // When the authorization status is OK, this object may include the headers that should be
+  // modified in the original request before it gets sent to the upstream.
```

Comment:
Is the desired behavior that the `headers` specified here would __ONLY__ modify existing headers in the original request? Could it be that the service could add new headers to the request.
Would it make sense for the authz server to send all the headers that should be sent upstream. So if `headers` is there then those are the only headers that would get forwarded to upstream? Or can the authz server not have enough context to do that.

Reply:
@saumoh Sorry for these confusing words. Yes, it should allow modifying the existing headers as well as inject new ones. I will rephrase all this comment. Thanks for pointing this out.